flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/cost/FlinkRelMdRowCount.scala

```diff
@@ -23,13 +23,17 @@ import org.apache.calcite.util.BuiltInMethod
 import org.apache.flink.table.plan.nodes.dataset.{DataSetCalc, DataSetSort}
 import java.lang.Double
 
+import org.apache.flink.table.rel.logical.FlinkLogicalCalc
+
 object FlinkRelMdRowCount extends RelMdRowCount {
 
-    val SOURCE: RelMetadataProvider = ReflectiveRelMetadataProvider.reflectiveSource(
-      BuiltInMethod.ROW_COUNT.method,
-      this)
+  val SOURCE: RelMetadataProvider = ReflectiveRelMetadataProvider.reflectiveSource(
+    BuiltInMethod.ROW_COUNT.method,
+    this)
+
+  def getRowCount(rel: FlinkLogicalCalc, mq: RelMetadataQuery): Double = rel.estimateRowCount(mq)
```

Comment:
Can we change this to `def getRowCount(rel: Calc, mq: RelMetadataQuery): Double` to cover `DataSetCalc` and `FlinkLogicalCalc` or would that be too generic?